agentops/time_travel.py

```diff
@@ -0,0 +1,137 @@
+                )
+        except FileNotFoundError:
+            return
+
+
+def fetch_time_travel_id(ttd_id):
+    try:
+        endpoint = environ.get("AGENTOPS_API_ENDPOINT", "https://api.agentops.ai")
+        payload = json.dumps({"ttd_id": ttd_id}).encode("utf-8")
+        ttd_res = HttpClient.post(f"{endpoint}/v2/get_ttd", payload)
+        if ttd_res.code != 200:
+            raise Exception(
+                f"Failed to fetch TTD with status code {ttd_res.status_code}"
+            )
+
+        prompt_to_returns_map = {
+            "completion_overrides": {
+                (
+                    str({"messages": item["prompt"]["messages"]})
+                    if item["prompt"].get("type") == "chatml"
+                    else str(item["prompt"])
+                ): item["returns"]
+                for item in ttd_res.body  # TODO: rename returns to completion_override
+            }
+        }
+        with open("agentops_time_travel.json", "w") as file:
+            json.dump(prompt_to_returns_map, file, indent=4)
+
+        set_time_travel_active_state("on")
+    except Exception as e:
+        manage_time_travel_state(activated=False, error=e)
+
+
+def fetch_completion_override_from_time_travel_cache(kwargs):
+    if not check_time_travel_active():
+        return
+
+    if TimeTravel()._completion_overrides_map:
+        search_prompt = str({"messages": kwargs["messages"]})
+        result_from_cache = TimeTravel()._completion_overrides_map.get(search_prompt)
+        return result_from_cache
+
+
+def fetch_prompt_override_from_time_travel_cache(kwargs):
+    if not check_time_travel_active():
+        return
+
+    if TimeTravel()._prompt_override_map:
+        search_prompt = str({"messages": kwargs["messages"]})
+        result_from_cache = TimeTravel()._prompt_override_map.get(search_prompt)
+        return json.loads(result_from_cache)
+
+
+def check_time_travel_active():
+    script_dir = os.path.dirname(os.path.abspath(__file__))
+    parent_dir = os.path.dirname(script_dir)
+    config_file_path = os.path.join(parent_dir, "agentops_time_travel.yaml")
+
+    with open(config_file_path, "r") as config_file:
+        config = yaml.safe_load(config_file)
+        if config.get("Time_Travel_Debugging_Active", True):
+            manage_time_travel_state(activated=True)
+            return True
+
+    return False
+
+
+def set_time_travel_active_state(active_setting):
+    config_path = "agentops_time_travel.yaml"
+    try:
+        with open(config_path, "r") as config_file:
+            config = yaml.safe_load(config_file) or {}
+    except FileNotFoundError:
+        config = {}
+
+    config["Time_Travel_Debugging_Active"] = True if active_setting == "on" else False
+
+    with open(config_path, "w") as config_file:
+        try:
+            yaml.dump(config, config_file)
+        except:
+            print(
+                f"🖇 AgentOps: Unable to write to {config_path}. Time Travel not activated"
```

Comment:
is there a reason to use print here and note logger?

Reply:
This happens outside of a Client session existing so no logger exists yet